public/app/plugins/panel/geomap/layers/data/markersLayer.tsx

```diff
@@ -112,43 +140,54 @@ export const markersLayer: MapLayerRegistryItem<MarkersConfig> = {
     return {
       init: () => vectorLayer,
       legend: legend,
-      update: (data: PanelData) => {
-        if (!data.series?.length) {
+      update: (newData: PanelData, previousData?: PanelData, updateDataquery?: (newDataquery: string) => void) => {
```

Comment:
rather than have the logic here... can it be up one layer?  That way each layer does not need to implement its own flavor?